web/app/themes/clarity/inc/admin/prior-party/prior-party-banner-email.php

```diff
@@ -0,0 +1,316 @@
+             */
+
+            foreach ($events as $post_id => $post_events) {
+                $post_type = get_post_type($post_id);
+                // Get the date to work out which banner is relevant.
+                $post_timestamp = get_the_date('U', $post_id);
+                // Get banner ref based on date
+                $post_banner = $this->getBannerByTimestamp($post_timestamp);
+                // Continue the loop if this post's banner doesn't match the one in the outer loop.
+                if (!isset($post_banner['reference']) ||  $post_banner['reference'] !== $banner['reference']) {
+                    continue;
+                }
+                // Don't count multiple events on the same post id. Just get the last one.
+                $last_action = end($post_events)['action'];
+                // Get the first event for comparison.
+                $first_action = $post_events[0]['action'];
+                // If the first event is opposite to the last event, then no change happened.
+                // e.g. if the first event is true then we can infer that the initial state was false.
+                // if the last event is not equal to the first event, last_action is true
+                // then there was no overall change.
+                if ($first_action !== $last_action) {
+                    continue;
+                }
+                if ($last_action === 'true') {
+                    $banner['post_types'][$post_type]['true_count']++;
+                } else {
+                    $banner['post_types'][$post_type]['false_count']++;
+                }
+            }
+
+            /**
+             * End a loop over the events.
+             */
+
+            // Get the number of changes for this banner.
+            $banner['change_count'] = array_reduce($banner['post_types'], fn ($c, $s) => $c + $s['true_count'] + $s['false_count'], 0);
+
+            // Build the url for reviewing.
+            $query_strings = array_merge($base_query_strings, ['ref' => $banner['reference']]);
+            $banner['review_url'] = admin_url('admin.php?' . http_build_query($query_strings));
+
+            $banner['email_body'] = $this->bannerArrayToText($banner);
+        }
+
+        /**
+         * End a loop over all the banners.
+         */
+
+
+        // Total changes for all banners.
+        $total_change_count = array_reduce($banners, fn ($c, $s) => $c + $s['change_count'], 0);
+
+        $email_heading = sprintf('<h2>Email digest for %s to %s</h2>', date('jS F, Y - g:i a', $from),  date('jS F, Y - g:i a', $to));
+        $email_bodies = array_map(fn ($b) => $b['email_body'], $banners);
+
+        $email = [
+            'subject' => sprintf('Moj Intranet - Prior Party Banner Digest %d recent changes', $total_change_count),
+            'body' => $email_heading . implode('<br/>', $email_bodies)
+        ];
+
+        return $email;
+    }
+
+    // TODO: schedule task for digest emails.
```

Comment:
@EarthlingDavey is this TODO for a later date?

Reply:
Cheers, I've got to come back to these.